### ISSUE TITLE
[GeoMechanicsApplication] Add support for non-uniform line loads

### ISSUE THREAD

**Issue by avdg81** [KratosMultiphysics/Kratos#11532]
As a user, I would like to be able to apply a non-uniform load in Kratos, such that I can model an embankment with it's actual shape.

**Description** 

To model an embankment in simulations with D-Settlement, we will use a non-uniform line load that represents the embankment's dead weight imposed on the subsoil. So far, we can only handle uniform line loads. However, due the embankment's geometry (with slopes) we actually need non-uniform line loads. The idea is to define a line load with a table, where the table describes the relationship between a coordinate on the subsoil (almost always X) and the magnitude of the line load. Since this is a different kind of dependency than what we have done so far (time dependencies only), we first need a way to retrieve the variables involved. This is part of another issue (see [PR #11528](https://github.com/KratosMultiphysics/Kratos/pull/11528)), which needs to be finished first.

[We have intentionally separated the changes needed in the Kratos core from the changes in the GeoMechanicsApplication.]

The following branch contains a first working implementation for this issue (no PR opened yet): https://github.com/KratosMultiphysics/Kratos/tree/geo/non-uniform-line-load-test

Identified implementation steps:
1. Change the reading of the table in GeoMechanics, such that it initializes the modelpart correctly from the mdpa file
2. Perform a simple test by directly integrating a simple non-uniform line load and comparing the results.

**Acceptance Criteria**

**Given** the user has modelled an embankment with its actual shape in Kratos,
**When** the user defines a 1D non-uniform line load for the simulation,
**Then** the application should correctly interpret and use this table for non-uniform load distribution.

**Given** the user has created a 1D non-uniform load,
**When** the user specifies the relationship between coordinates and line load magnitudestable as input in a table,
**Then** the load table generated in the mdpa file, should be containing the relationship between a coordinate (almost always X) and the magnitude of the line load.
